src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -909,6 +929,29 @@ abstract class ManagedStrategy(
         }
     }
 
+    /**
+     * Tracks a specific field of an [obj], if the [obj] is either `null` (which means that field is static),
+     * or one this objects which contains it is already stored.
+     */
+    fun updateSnapshotOnFieldAccess(obj: Any?, className: String, fieldName: String) = runInIgnoredSection {
+        staticMemorySnapshot.trackField(obj, className, fieldName)
+    }
+
+    /**
+     * Tracks a specific [array] element at [index], if the [array] is already tracked.
+     */
+    fun updateSnapshotOnArrayElementAccess(array: Any, index: Int) = runInIgnoredSection {
```

Comment:
"access" or "update"? (the same for other functions)

Reply:
We need to invoke this method for both reads and writes to the array elements (same for other functions), so the correct name would be `*Access(...)`